modules/miner/update.go

```diff
@@ -60,3 +34,26 @@ func (m *Miner) ReceiveTransactionPoolUpdate(cc modules.ConsensusChange, unconfi
 	m.target = target
 	m.earliestTimestamp = timestamp
 }
+
+// ReceiveUpdatedUnconfirmedTransactions will replace the current unconfirmed
+// set of transactions with the input transactions. This is a part of the
+// TransactionPoolSubscriber interface.
+func (m *Miner) ReceiveUpdatedUnconfirmedTransactions(unconfirmedTransactions []types.Transaction, _ modules.ConsensusChange) {
+	lockID := m.mu.Lock()
+	defer m.mu.Unlock(lockID)
+
+	m.transactions = nil
+	remainingSize := int(types.BlockSizeLimit - 5e3)
+	for {
+		if len(unconfirmedTransactions) == 0 {
+			break
+		}
+		remainingSize -= len(encoding.Marshal(unconfirmedTransactions[0]))
+		if remainingSize < 0 {
+			break
+		}
+
+		m.transactions = append(m.transactions, unconfirmedTransactions[0])
+		unconfirmedTransactions = unconfirmedTransactions[1:]
+	}
```

Comment:
is there a reason for not using a traditional range loop here? It's not clear to me.